scripts/sign.py

```diff
@@ -32,27 +31,24 @@ def main():
     from Crypto.Signature import PKCS1_v1_5
     from Crypto.Hash import SHA256
     from Crypto.PublicKey import RSA
-    import struct
+    import base64, os.path, struct
 
     args = get_args()
 
-    f = open(args.key, 'rb')
-    key = RSA.importKey(f.read())
-    f.close()
+    with open(args.key, 'rb') as f:
+        key = RSA.importKey(f.read())
 
-    f = open(args.inf, 'rb')
-    img = f.read()
-    f.close()
+    with open(args.inf, 'rb') as f:
+        img = f.read()
 
-    signer = PKCS1_v1_5.new(key)
     h = SHA256.new()
 
     digest_len = h.digest_size
-    sig_len = len(signer.sign(h))
+    sig_len = (key.size() + 1) / 8
```

Comment:
Not necessarily, see https://www.dlitz.net/software/pycrypto/api/current/Crypto.PublicKey.RSA._RSAobj-class.html#size 
> Tell the maximum number of bits that can be handled by this key. 
Which is $n-1$ for a proper $n$-bit RSA key.
Unless we have a modulus length that is not a multiple of 8, we do not need a proper ceiling function here.

Reply:
Sorry, I don't see the reference to `key.size()` being the key bit size minus 1.

Regarding the change, I think it is better for the script to nicely round bit size into byte size, whatever is the input bit size value (modulo 8 or not). @jforissier, your thoughts?

Reply:
What happens if key size is not a multiple of 8? Do things even work?

Reply:
@etienne-lms The maximum number of bits that can be handled by a key is determined by the size of the modulus in bits. It must be strictly smaller, as otherwise the modulus would be all ones. On the other hand, if the difference was more than one, e.g. two, that would rule out about half of the posible bitstrings. Again, not something I would call a key of proper size.

@jforissier from the cryptographic point of view you can have something like a 2047-bit RSA, though most software will not support it.

Reply:
> What happens if key size is not a multiple of 8? Do things even work?

I think this matters the Crypto module, not this script.
If by construction `key.size()` value is a multiple of 8, then the current operation is fine but could be changed to:
```patch
-    sig_len = (key.size() + 1) / 8
+    sig_len = key.size() / 8
```
I still think `sig_len = (key.size() + 7) / 8` is better here.

Reply:
> If by construction `key.size()` value is a multiple of 8, 

It is, by construction, 1 less than a multiple of 8.

Reply:
Indeed, looks like it is `some_rsa_key.size()` returns the key bit size minus 1, but I could not find a place that documents this.
Note `some_rsa_key.n.bit_length()` return the exact bit size.
Maybe `sig_len = (key.n.bit_length() + 7) / 8`.

If choosing `(key.size() + 1) / 8`, maybe add a comment stating expected `key.size()` value.

Reply:
> Indeed, looks like it is `some_rsa_key.size()` returns the key bit size minus 1, but I could not find a place that documents this.

https://github.com/dlitz/pycrypto/blob/master/lib/Crypto/PublicKey/_RSA.py#L80

The proper way would be ` Crypto.Util.number.ceil_div(Crypto.Util.number.size(key.n),8)`, see https://github.com/dlitz/pycrypto/blob/master/lib/Crypto/Signature/PKCS1_v1_5.py#L110

But the attribute `n` is not documented and thus I prefer to use the documented API, which means I'll be changing this to ` Crypto.Util.number.ceil_div(key.size() + 1, 8)` in the next update of the PR.

(btw: generating an RSA key with pycrypto enforces a multiple of 256 bits)

Reply:
thanks for the ref. Maybe a word of comment would help next readers to understand to computation.

By the way, what was wrong with the original one 'sign_len = len(signer.sign(h))'?

Reply:
> By the way, what was wrong with the original one 'sign_len = len(signer.sign(h))'?

This would have thrown an exception in the case of offline signing.

Reply:
Indeed, right! sorry.